File: api/node/src/models/Team.ts
import Player from "./Player";

export default class Team {
    players: Array<Player>;
    score: number;

    constructor(players: Array<Player>) {
        this.players = players;
        this.score = 0;
        this.players.map(player => {
            this.score += player.score
        })
    }
}

File: api/go/models/team.go
package models

type Team struct {
	Players []Player `json:"players"`
	Score   int      `json:"score"`
}

func NewTeam(players []Player) Team {
	score := 0
	for _, player := range players {
		score += player.Score
	}
	return Team{Players: players, Score: score}
}
